pkg/scheduler/plugins/coscheduling/util/gang_helper.go

```diff
@@ -56,11 +57,23 @@ func IsPodNeedGang(pod *v1.Pod) bool {
 	return GetGangNameByPod(pod) != ""
 }
 
-func ParsePgTimeoutSeconds(timeoutSeconds int32) (time.Duration, error) {
-	if timeoutSeconds <= 0 {
-		return 0, fmt.Errorf("podGroup timeout value is illegal,timeout Value:%v", timeoutSeconds)
+// GetWaitTimeDuration returns a wait timeout based on the following precedences:
+// 1. spec.scheduleTimeoutSeconds of the given pg, if specified
+// 2. given defaultTimeout, if not nil
+// 3. fall back to 0
+func GetWaitTimeDuration(pg *v1alpha1.PodGroup, defaultTimeout *metav1.Duration) time.Duration {
+	if pg != nil && pg.Spec.ScheduleTimeoutSeconds != nil &&
+		*pg.Spec.ScheduleTimeoutSeconds > 0 {
+		return time.Duration(*pg.Spec.ScheduleTimeoutSeconds) * time.Second
 	}
-	return time.Duration(timeoutSeconds) * time.Second, nil
+
+	klog.Errorf("podGroup timeout value is nil or illegal, podGroupName: %v", pg.Name)
```

Comment:
It is not an error not to provide this value if we set the default. So this should be
```
if defaultTimeout != nil {
    return defaultTimeout.Duration
} else {
    klog.Errorf("podGroup timeout value is nil or illegal, podGroupName: %v", pg.Name)
}
```

Reply:
Done